src/PreparedWrites.cc

```diff
@@ -0,0 +1,521 @@
+/* Copyright (c) 2015 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include "UnackedRpcResults.h"
+#include "LeaseCommon.h"
+#include "MasterService.h"
+#include "ObjectManager.h"
+
+namespace RAMCloud {
+
+/**
+ * Construct a PreparedOp in preparation for storing it in the log.
+ * This form is used when the header information is available in
+ * individual pieces, while the keys and data are stored in  a Buffer
+ * (typical use: during txPrepare RPCs).
+ *
+ * \param type
+ *      Type of the staged operation.
+ * \param clientId
+ *      leaseId given for this linearizable RPC.
+ * \param rpcId
+ *      rpcId given for this linearizable RPC.
+ * \param participantCount
+ *      Number of objects participating current transaction.
+ * \param participants
```

Comment:
This pointer appears to get copied directly into the PreparedOp object. Doesn't this create opportunities for dangling pointers and memory corruption? . Let's discuss. If nothing else, I think you need to document that the caller must ensure that the lifetime of participants exceeds the lifetime of this object.